src/main.rs

```diff
@@ -366,8 +375,8 @@ fn versioned_cargo_exec_on_packages(
     cx: &Context,
     packages: &[PackageRuns<'_>],
     cargo_version: u32,
-    progress: &mut Progress,
-    keep_going: &mut KeepGoing,
+    progress: &Arc<Mutex<Progress>>,
+    keep_going: &Arc<Mutex<KeepGoing>>,
```

Comment:
Here (and every other `&Arc<Mutex<T>>`), I think it would make sense to hold an `Arc<Mutex<T>>` directly. Cloning each time is incredibly cheap (incrementing an atomic), which is better than passing around a double pointer.